modules/services/fb_streetview_ai_suggestions.js

```diff
@@ -0,0 +1,499 @@
+        };
+    }
+    return members;
+}
+
+
+var parsers = {
+    node: function nodeData(obj, uid) {
+        var attrs = obj.attributes;
+        return new osmNode({
+            id: uid,
+            visible: getVisible(attrs),
+            loc: getLoc(attrs),
+            tags: getTags(obj)
+        });
+    },
+
+    way: function wayData(obj, uid) {
+        var attrs = obj.attributes;
+        return new osmWay({
+            id: uid,
+            suggestionId: obj.getAttribute("suggestion-id"),
+            visible: getVisible(attrs),
+            tags: getTags(obj),
+            nodes: getNodes(obj),
+        });
+    },
+
+    relation: function relationData(obj, uid) {
+        var attrs = obj.attributes;
+        return new osmRelation({
+            id: uid,
+            suggestionId: obj.getAttribute("suggestion-id"),
+            visible: getVisible(attrs),
+            tags: getTags(obj),
+            members: getMembers(obj)
+        });
+    },
+};
+
+
+// parse <osm-road> element inside <sidewalk-suggestion>
+function praseOsmRoadMeta(xmlEle) {
+    var osmRoadMeta = {};
+    osmRoadMeta.wid = osmEntity.id.fromOSM("way", xmlEle.getAttribute("way-id"));
+    osmRoadMeta.version = xmlEle.getAttribute("version");
+    osmRoadMeta.sidewalk_tag = xmlEle.getAttribute("sidewalk-tag");
+
+    var nodeEles = xmlEle.getElementsByTagName("nd");
+    var nodes = new Array(nodeEles.length);
+    for (var i = 0; i < nodeEles.length; i++) {
+        var n = nodeEles[i];
+        nodes[i] = {
+            nid: osmEntity.id.fromOSM("node", n.getAttribute("ref")),
+            version: n.getAttribute("version"),
+            lat: parseFloat(n.getAttribute("lat")),
+            lon: parseFloat(n.getAttribute("lon"))
+        };
+    }
+    osmRoadMeta.nodes = nodes;
+    return osmRoadMeta;
+}
+
+
+// parse <steet-view-image-set> element inside <sidewalk-suggestion>
+function praseStreetViewImageSet(xmlEle) {
+    var streetViewImageSet = {};
+    streetViewImageSet.id = xmlEle.getAttribute("id");
+    streetViewImageSet.cameraPointingDirection = xmlEle.getAttribute("camera-pointing-direction");
+    streetViewImageSet.relativeSideToRoad = xmlEle.getAttribute("relative-side-to-road");
+
+    var imageEles = xmlEle.getElementsByTagName("street-view-image");
+    var images = new Array(imageEles.length);
+    for (var i = 0; i < imageEles.length; i++) {
+        var img = imageEles[i];
+        images[i] = {
+            key: img.getAttribute("key"),
+            url: img.getAttribute("url"),
+            sidewalkSide: img.getAttribute("sidewalk-side"),
+            lat: parseFloat(img.getAttribute("lat")),
+            lon: parseFloat(img.getAttribute('lon'))
+        };
+    }
+    streetViewImageSet.images = images;
+    return streetViewImageSet;
+}
+
+
+// parse <sidewalk-suggestion>
+function praseSidewalkSuggestion(xmlEle, cubitorContext) {
+    var id = xmlEle.getAttribute("id");
+    if (cubitorContext[id]) return;
```

Comment:
I'm a little curious on the reasoning for this; would we not want to update the context on the off chance that the local data is stale?

Reply:
trying to make this simple for v0 since iD is doing `skipSeen` for OSM entities anyway
https://github.com/facebookincubator/RapiD/blob/streetview_workflow/modules/services/osm.js#L411-L442

I'll add TODO comment about making it comprehensive later